### PR TITLE
[markdown] Fix linting rule 'no-heading-punctuation' 

### PR DESCRIPTION
**Pull request by scinos**
### Background

After linting our Markdown files following prettier format, I'd like to introduce content-based rules from `remark`. The plugin we use (`eslint-plugin-md`) recommends [Markdown Style Guide](https://cirosantilli.com/markdown-style-guide/), and it has [a preset](https://www.npmjs.com/package/remark-preset-lint-markdown-style-guide) for it.

Unfortunately this preset doesn't have the capacity of auto-fix the problems. Instead of enabling the preset and having a ton of errors, I'll enable the rules of the preset one by one and fix each error. When all the rules are enabled we can switch the hardcoded list of rules by the preset.

### Changes

Enables rule [no-heading-punctuation](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-heading-punctuation). This rule forbids to use `.` or `:` at the end of the titles (https://cirosantilli.com/markdown-style-guide/#punctuation-at-the-end-of-headers)

```
Good:

# Foo
# Bar

Bad:

# Foo:
# Bar.
```

### Testing
Run `./node_modules/.bin/eslint --ext .md .` and check there are no errors

### CODE REVIEW

File: client/layout/guided-tours/docs/examples/README.md [hunk 1]
```diff
@@ -10,6 +10,6 @@ When we archive an unused tour as an example, we aim to move all the selectors o
 
 - `simple-payments-end-of-year-guide.js`: This tour was used as a temporary end-of-year advertisement for the Simple Payments feature. It triggers on a variety of paths but only for premium and business users. The tour guides people through the creation of a page and the insertion of a Simple Payments button.
 
-## Selectors etc.
+## Selectors etc
```
Comment (saramarcondes) [inline client/layout/guided-tours/docs/examples/README.md:13]: Typically `etc.` does include a `.` at the end of it as it is an abbreviation. Should we ignore these cases?

**Review by sarayourfriend — APPROVED** [review]
LGTM. I think we should consider allowing `etc.` as it is stylistically correct to include a period in that case, but otherwise this PR is ready to merge IMO.

File: client/layout/guided-tours/docs/examples/README.md (hunk 1 above)
Comment (scinos) [reply client/layout/guided-tours/docs/examples/README.md:13]: Very good point. The rule https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-heading-punctuation doesn't allow lots of configuration, just a list of characters to ban. The plugin uses `:.`, but maybe we should just use `:`

Comment (scinos) [reply client/layout/guided-tours/docs/examples/README.md:13]: I figured out how we can use something `<!--eslint ignore no-heading-punctuation-->` to disable specific rules for the next line, I'll use that to enable `etc.`

**Pull request merged**